### PR TITLE
Add test for invalid macaroon

### PR DESCRIPTION
**Pull request by DhananjayPurohit**

### CODE REVIEW

File: .env_example [hunk 1]
```diff
@@ -9,5 +9,8 @@ LN_CLIENT_TYPE=
 # Root key for minting macaroons
 ROOT_KEY=
 
-TEST_MACAROON=
-TEST_PREIMAGE=
+TEST_MACAROON_VALID=
```
Comment (kiwiidb) [inline .env_example:12]: I think it is better to just hard-code these macaroon values, they are not sensitive info.

File: ginlsat/ginlsat.go [hunk 2]
```diff
@@ -125,15 +124,13 @@ func (lsatmiddleware *GinLsatMiddleware) SetLSATHeader(c *gin.Context) {
...
-		c.Error(err)
 		c.Set("LSAT", &LsatInfo{
 			Error: err,
 		})
 		return
 	}
 	macaroonString, err := macaroonutils.GetMacaroonAsString(paymentHash)
 	if err != nil {
-		c.Error(err)
 		c.Set("LSAT", &LsatInfo{
```
Comment (kiwiidb) [inline ginlsat/ginlsat.go:134]: Shouldn't we add a type: "ERROR" here or something?

File: ginlsat_test.go [hunk 3]
```diff
@@ -98,10 +100,16 @@ func TestLsatWithLNURLConfig(t *testing.T) {
 	err := godotenv.Load(".env")
 	assert.NoError(t, err)
 
+	LNURL_ADDRESS := os.Getenv("LNURL_ADDRESS")
+	TEST_MACAROON_VALID := os.Getenv("TEST_MACAROON_VALID")
```
Comment (kiwiidb) [inline ginlsat_test.go:104]: See above, better to hard-code the macaroon/preimages here.

**Review by kiwiidb — CHANGES_REQUESTED** [review]

File: ginlsat/ginlsat.go (hunk 2 above)
Comment (DhananjayPurohit) [reply ginlsat/ginlsat.go:134]: We have already put the error into LSAT info that can be served along with the request response (if any error occurred). I found it a bit irrelevant when the error got logged with the test results for the invalid macaroon test. I can undo the [commit ](https://github.com/getAlby/gin-lsat/pull/17/commits/e6e5a60b3d22da757c757ef4d34bcc7538dc509d) or even we can implement a logging mechanism that can put all the generated errors with the timestamp in a separate file.

**Review by kiwiidb — APPROVED** [review]

**Pull request merged**